apis/python/src/tiledbsoma/types.py

```diff
@@ -31,3 +32,21 @@
 
 SparseCoordinates = Union[int, slice, Tuple[int, ...], List[int], pa.IntegerArray]
 SparseNdCoordinates = Tuple[DenseCoordinates, ...]
+
+# Acceptable ways to index:
+# * None
+# * A sequence of coordinates is accepted, one per dimension.
+# * Sequence length must be at least one and <= number of dimensions.
+# * If the sequence contains missing coordinates (length less than number of dimensions),
+#   then "slice(None)" is assumed for the missing dimensions.
+# * Per-dimension, explicitly specified coordinates can be one of: None, a value, a
+#   list/ndarray/pyarrow.Array/etc of values, a slice, etc.
+
+# TODO: add support for non-ints once the libtiledbsoma SOMAReader class has supports
+# for non-ints. See also:
+# https://github.com/single-cell-data/TileDB-SOMA/issues/418
+# https://github.com/single-cell-data/TileDB-SOMA/issues/419
+SparseIndexedDataFrameCoordinate = Optional[
```

Comment:
IMHO, this would be clearer if defined as explicitly accepting a None, i.e.
>     Union[None, int, slice, Sequence[int], pa.Array, pa.ChunkedArray, np.ndarray]

No change to the meaning, just clarity

Reply:
You are correct as far as the typing goes, but the implication of `Optional` is that the value can be omitted.  That is _not_ the case here -- the value can never be omitted, but it can be a None.  Quite different semantics.

I'll put it another way, index values are not optional, as they are _positional_.

Reply:
OK @bkmartinjr I'll do as you ask; I've gotten feedback in the past from more Pythonic devs to use `Optional[X]` when what I meant was `Union[None, X]`. It's your review. :)

Reply:
https://github.com/single-cell-data/TileDB-SOMA/pull/483/commits/d0edba8d5bebc0337179359642d6b1b2d7bba9b9

Reply:
Can i suggest you consult with Paul or Isaiah - I am not an authority, just giving you feedback about how I read the spec :-)